src/app/services/transmart-resource/transmart-mapper.ts

```diff
@@ -175,9 +183,14 @@ export class TransmartMapper {
       // get row dimensions
       transmartTable.rows[i].dimensions.forEach((inRowDim: TransmartInRowDimension) => {
         if (inRowDim.key == null) {
```

Comment:
try to use === instead of ==

Reply:
this was on purpose, because `==` checks both `null` and `undefined`

Reply:
in a current data-table json representation the dimension object in a row has name and either a key or an element. Key means that more details are described in dimension description part, while the element means that the dimension is inline - has one value (eg. start time dimension) so there is no need to add a key there, easier to just put inline value. 

So either key is undefined or element. There is also an option that both are null (eg. some observation does not have a start time or end time).

As I understand from TypeScript guides, '==' was meant to be used instead of '===null or ===undefined', but if it is less readable for you, I can change it.